CMakeLists.txt

```diff
@@ -523,6 +524,33 @@ if(OPENCV_DISABLE_FILESYSTEM_SUPPORT)
   add_definitions(-DOPENCV_HAVE_FILESYSTEM_SUPPORT=0)
 endif()
 
+if(OPENCV_DISABLE_THREAD_SUPPORT)
+  add_definitions(-DOPENCV_DISABLE_THREAD_SUPPORT)
```

Comment:
The option is not exported to `OpenCV-config.cmake` and other build configs. All methods in header files will use another pre-processor branch and break user code.

Reply:
I am happy to do the export of the variable but I am not sure I understand the issue.

Also, how can I export `OPENCV_DISABLE_THREAD_SUPPORT` in `OpenCV-config.cmake`? I have looked a bit into `cmake/OpenCVGenConfig.cmake` but I am missing the mechanism that does this export.